### ISSUE TITLE
Adding -a=b or --a=b causes a "hGetContents: invalid argument (invalid byte sequence)" error

### ISSUE THREAD

**Issue by ferrarimarco** [hadolint/hadolint#152]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

`RUN npm install -–progress=false` should not raise any error

### Actual behavior

`Dockerfile: hGetContents: invalid argument (invalid byte sequence)`

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:** `Haskell Dockerfile Linter v1.2.6-0-ge03cbb0`

```bash
Dockerfile: hGetContents: invalid argument (invalid byte sequence)
```

**Comment by zemanlx** [comment]
@ferrarimarco you gave me a tough time :smile:  I have copied `RUN npm install -–progress=false` to a new test. It was failing, even I have put a fix in the code. And after a few minutes of staring at code why it is not working, I have noticed that second dash `-` is not dash.

BTW fix is on its way.

**Comment by ferrarimarco** [comment]
OMG sorry! I copied and pasted it from SO 😢
